tests/store/test_runs_artifact_repo.py

```diff
@@ -0,0 +1,47 @@
+import mock
+import pytest
+
+from mlflow.exceptions import MlflowException
+from mlflow.store.runs_artifact_repo import RunsArtifactRepository
+
+
+@pytest.mark.parametrize("uri, expected_run_id, expected_artifact_path", [
+    ('runs:/1234abcdf1394asdfwer33/path/to/model', '1234abcdf1394asdfwer33', 'path/to/model'),
+    ('runs:/1234abcdf1394asdfwer33/path/to/model/', '1234abcdf1394asdfwer33', 'path/to/model/'),
+    ('runs:/1234abcdf1394asdfwer33', '1234abcdf1394asdfwer33', None),
+    ('runs:/1234abcdf1394asdfwer33/', '1234abcdf1394asdfwer33', None),
+    ('runs:///1234abcdf1394asdfwer33/', '1234abcdf1394asdfwer33', None),
```

Comment:
@mparkhe do you mind confirming this URI *should* be accepted as a valid Runs URI?

Reply:
i think this one is valid. wiki link attached.

Reply:
It's a minor point, but I don't think that this necessarily *has* to be valid. As far as I can tell, each URI type determines the constraints of the `authority` field; `file` URIs allow the `authority` field to be an empty string, automatically resolving the empty field to `localhost` by default. In contrast, `http` URIs do not allow the `authority` field to be empty. For reference, here's a relevant GitHub thread from the `curl` project: https://github.com/curl/curl/issues/791.

Reply:
Thanks for the pointer! Yeah I think the runs URIs are more like file URIs in that it could be local or remote. It'll be more relevant when we introduce hostnames to the runs URI but I think the current version is consistent with that future.